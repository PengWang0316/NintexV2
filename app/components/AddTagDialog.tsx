import React, {
  useState, memo, ReactElement, ChangeEvent,
} from 'react';
import { connect } from 'react-redux';
import {
  Button, Dialog, DialogActions, DialogContent,
  DialogContentText, DialogTitle, Chip, TextField,
} from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import I18n from '@kevinwang0316/i18n';
import { CirclePicker } from 'react-color';

import { addTag as addTagAction } from '../store/Tags/actions';
import { TagsType } from '../store/Tags/types';
import { AppState } from '../store/ConfigureStore';

interface Props {
  open: boolean;
  tags: TagsType | null;
  addTag: (content: string, color: string) => void;
  handleClose: (event: React.MouseEvent) => void;
}

const useStyles = makeStyles({
  flexDiv: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-between',
    marginTop: 20,
  },
  secondContentText: { marginTop: 30 },
  btnDiv: { textAlign: 'right' },
});

const DEFAULT_COLOT = '#f44336';
const COLOR_REGEXP = /^#[\d\w]{6}$/;

export const AddTagDialog = ({
  open, handleClose, tags = null, addTag,
}: Props): ReactElement => {
  const classes = useStyles({});
  const [tagColor, setTagColor] = useState(DEFAULT_COLOT);
  const [tagText, setTagText] = useState('');

  const handleChangeComplete = color => setTagColor(color.hex);

  const handleInputChange = (event: ChangeEvent<HTMLInputElement>) => setTagText(event.target.value);

  const handleAddBtnClick = () => {
    // Do a validation here and also do it in the backend again
    if (tagText && tagText !== '' && tagColor.match(COLOR_REGEXP)) {
      addTag(tagText, tagColor);
      setTagText('');
      setTagColor(DEFAULT_COLOT);
    }
  };

  return (
    <Dialog
      open={open}
      keepMounted
      maxWidth="md"
      onClose={handleClose}
      aria-labelledby="alert-dialog-slide-title"
      aria-describedby="alert-dialog-slide-description"
    >
      <DialogTitle id="alert-dialog-slide-title">{I18n.get('addTagDialogTitle')}</DialogTitle>
      <DialogContent>
        <DialogContentText id="alert-dialog-slide-description">
          {I18n.get('addTagDialogExist')}
        </DialogContentText>
        {tags && (
          <div>
            {Object.keys(tags).map(tag => (
              <Chip
                label={`${tags[tag][0]}`}
                key={`${tags[tag][0]}${tags[tag][1]}`}
                clickable
                style={{ backgroundColor: `${tags[tag][1]}`, margin: '0 10px 5px 0', color: 'white' }}
              />
            ))}
          </div>
        )}
        <DialogContentText className={classes.secondContentText}>
          {I18n.get('addTagDialogContent')}
        </DialogContentText>
        <div className={classes.flexDiv}>
          <div>
            <div>
              <TextField
                label={I18n.get('tagContentTitle')}
                value={tagText}
              // className={classes.textField}
                margin="normal"
                onChange={handleInputChange}
              />
            </div>
            <div className={classes.btnDiv}>
              <Button size="small" variant="outlined" onClick={handleAddBtnClick} color="secondary">
                {I18n.get('add')}
              </Button>
            </div>

          </div>
          <div><CirclePicker onChangeComplete={handleChangeComplete} color={tagColor} /></div>
        </div>
      </DialogContent>
      <DialogActions>
        <Button onClick={handleClose} color="primary">
          {I18n.get('close')}
        </Button>
      </DialogActions>
    </Dialog>
  );
};

const mapStateToProps = (state: AppState) => ({
  tags: state.tags,
});
const mapDispatchToProps = dispatch => ({
  addTag: (content: string, color: string) => dispatch(addTagAction(content, color)),
});
export default connect(mapStateToProps, mapDispatchToProps)(memo(AddTagDialog));
