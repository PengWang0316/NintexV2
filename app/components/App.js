import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles';
import importedComponent from 'react-imported-component';
import CssBaseline from '@material-ui/core/CssBaseline';
import Amplify from 'aws-amplify';
import { withAuthenticator } from 'aws-amplify-react';

import Navbar from './Navbar';
import MenuDrawer from './MenuDrawer';
import { checkInstanceStatus as checkInstanceStatusAction } from '../actions/InstanceActions';
import {
  HOME_PAGE_URL, WORKFLOW_MANAGER_PAGE_URL, SIGNIN_PAGE_URL,
  cognitoConfig, amplifyAuthSignOption, MONITOR_CENTER_PAGE_URL,
} from '../config';
import LoadingAnimation from './SharedComponents/LoadingAnimation';

Amplify.configure({
  Auth: cognitoConfig,
});

const theme = createMuiTheme({
  typography: {
    useNextVariants: true,
  },
  palette: {
    primary: {
      light: '#4A6572',
      main: '#344955',
      dark: '#232F34',
      contrastText: '#fff',
    },
    secondary: {
      light: '#497ca4',
      main: '#105075',
      dark: '#002884',
      contrastText: '#fff',
    },
  },
});

/* istanbul ignore next */
const HomePage = importedComponent(() => import(/* webpackChunkName: "HomePageContainer" *//* webpackPrefetch: true */ './containers/HomePageContainer').catch(err => console.log(err)), { LoadingComponent: LoadingAnimation });
const ManagementPage = importedComponent(() => import(/* webpackChunkName: "ManagementContainer" *//* webpackPrefetch: true */ './containers/ManagementContainer').catch(err => console.log(err)), { LoadingComponent: LoadingAnimation });
const MonitorPage = importedComponent(() => import(/* webpackChunkName: "MonitorContainer" *//* webpackPrefetch: true */ './containers/MonitorContainer').catch(err => console.log(err)), { LoadingComponent: LoadingAnimation });

const INTERVAL_TIME = 5 * 60 * 1000;
let autoMonitorJob = null;

/**
 * The root component that contains the theme, routers, navbar, and login dialog
 */
export const App = ({ monitorList, checkInstanceStatus }) => {
  useEffect(() => {
    if (Object.keys(monitorList.data).length !== 0) {
      if (autoMonitorJob) clearInterval(autoMonitorJob);
      autoMonitorJob = setInterval(() => checkInstanceStatus(monitorList.data), INTERVAL_TIME);
      // console.log('item changed, start a new job');
    } else if (autoMonitorJob && Object.keys(monitorList.data).length === 0) {
      if (autoMonitorJob) clearInterval(autoMonitorJob);
      autoMonitorJob = null;
      // console.log('empty list, stop job');
    }
  });
  return (
    <MuiThemeProvider theme={theme}>
      <Router>
        <div>
          <CssBaseline />
          <Navbar />
          <MenuDrawer />
          <main>
            <Switch>
              <Route exact path={HOME_PAGE_URL} component={HomePage} />
              <Route exact path={WORKFLOW_MANAGER_PAGE_URL} component={ManagementPage} />
              <Route exact path={MONITOR_CENTER_PAGE_URL} component={MonitorPage} />
              <Route exact path={SIGNIN_PAGE_URL} component={withAuthenticator(HomePage, amplifyAuthSignOption)} />
              <Route render={() => <p>Not Fount!</p>} />
            </Switch>
          </main>
        </div>
      </Router>
    </MuiThemeProvider>
  );
};
App.propTypes = {
  monitorList: PropTypes.objectOf(PropTypes.any).isRequired,
  checkInstanceStatus: PropTypes.func.isRequired,
};
const mapStateToProps = state => ({ ...state });
const mapDispatchToProps = {
  checkInstanceStatus: checkInstanceStatusAction,
};
export default connect(mapStateToProps, mapDispatchToProps)(App);
