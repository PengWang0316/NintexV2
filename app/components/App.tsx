import React, { Suspense, lazy } from 'react';
import { connect } from 'react-redux';
// import PropTypes from 'prop-types';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles';
import CssBaseline from '@material-ui/core/CssBaseline';
import Amplify from 'aws-amplify';
import { withAuthenticator } from 'aws-amplify-react';

import Navbar from './Navbar';
import MenuDrawer from './MenuDrawer';
import { checkInstanceStatus as checkInstanceStatusAction } from '../store/MonitorList/actions';
import {
  HOME_PAGE_URL, WORKFLOW_MANAGER_PAGE_URL, SIGNIN_PAGE_URL,
  cognitoConfig, amplifyAuthSignOption, MONITOR_CENTER_PAGE_URL,
} from '../config';
import { MonitorListType, MonitorListData } from '../store/MonitorList/types';

Amplify.configure({
  Auth: cognitoConfig,
});

const theme = createMuiTheme({
  palette: {
    primary: {
      light: '#4A6572',
      main: '#344955',
      dark: '#232F34',
      contrastText: '#fff',
    },
    secondary: {
      light: '#497ca4',
      main: '#105075',
      dark: '#002884',
      contrastText: '#fff',
    },
  },
});

/* istanbul ignore next */
const HomePage = lazy(() => import(/* webpackChunkName: "HomePageContainer" *//* webpackPrefetch: true */ './containers/HomePageContainer'));
/* istanbul ignore next */
const ManagementPage = lazy(() => import(/* webpackChunkName: "ManagementContainer" *//* webpackPrefetch: true */ './containers/ManagementContainer'));
/* istanbul ignore next */
const MonitorPage = lazy(() => import(/* webpackChunkName: "MonitorContainer" *//* webpackPrefetch: true */ './containers/MonitorContainer'));

const INTERVAL_TIME = 1 * 60 * 1000;
let autoMonitorJob = null;
interface PropType {
  monitorList: MonitorListType;
  checkInstanceStatus: (monitoredWorkflows: MonitorListData) => void;
}
/**
 * The root component that contains the theme, routers, navbar, and login dialog
 */
export const App = ({ monitorList, checkInstanceStatus }: PropType) => {
  React.useEffect(() => {
    if (Object.keys(monitorList.data).length !== 0) {
      // item changed, start a new job
      if (autoMonitorJob) clearInterval(autoMonitorJob);
      autoMonitorJob = setInterval(() => checkInstanceStatus(monitorList.data), INTERVAL_TIME);
    } else if (autoMonitorJob && Object.keys(monitorList.data).length === 0) {
      // Stop the fetching job
      if (autoMonitorJob) clearInterval(autoMonitorJob);
      autoMonitorJob = null;
    }
  });
  return (
    <MuiThemeProvider theme={theme}>
      <Router>
        <div>
          <CssBaseline />
          <Navbar />
          <MenuDrawer />
          <main>
            <Suspense fallback={<div>Loading...</div>}>
              <Switch>
                <Route exact path={HOME_PAGE_URL} component={HomePage} />
                <Route exact path={WORKFLOW_MANAGER_PAGE_URL} component={ManagementPage} />
                <Route exact path={MONITOR_CENTER_PAGE_URL} component={MonitorPage} />
                <Route exact path={SIGNIN_PAGE_URL} component={withAuthenticator(HomePage, amplifyAuthSignOption)} />
                <Route render={() => <p>Not Fount!</p>} />
              </Switch>
            </Suspense>
          </main>
        </div>
      </Router>
    </MuiThemeProvider>
  );
};
/* istanbul ignore next */
const mapStateToProps = ({ monitorList }: { monitorList: MonitorListType }) => ({ monitorList });
/* istanbul ignore next */
const mapDispatchToProps = {
  checkInstanceStatus: checkInstanceStatusAction,
};
export default connect(mapStateToProps, mapDispatchToProps)(App);
