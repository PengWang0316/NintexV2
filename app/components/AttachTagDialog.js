import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import {
  Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle, Slide, Chip,
} from '@material-ui/core';
import I18n from '@kevinwang0316/i18n';

import { updateTagFromWorkflow as updateTagFromWorkflowAction } from '../store/Workflows/actions';

const Transition = React.forwardRef((props, ref) => <Slide direction="down" ref={ref} {...props} />);

export const AttachTagDialog = ({
  isOpen, handleClose, tags, workflowId, workflows, updateTagFromWorkflow,
}) => {
  const existedTagIds = Object.keys(workflows.data).length !== 0 && workflowId && workflows.data[workflowId].tags ? workflows.data[workflowId].tags.split(',') : [];
  const handleTagClick = event => updateTagFromWorkflow(
    workflowId,
    `${workflows.data[workflowId].tags ? `${workflows.data[workflowId].tags},` : ''}${event.target.tagName === 'span' ? event.target.parentNode.parentNode.getAttribute('name') : event.target.parentNode.getAttribute('name')}`,
  );


  return (
    <Dialog
      open={isOpen}
      TransitionComponent={Transition}
      keepMounted
      maxWidth="md"
      onClose={handleClose}
      aria-labelledby="alert-dialog-slide-title"
      aria-describedby="alert-dialog-slide-description"
    >
      <DialogTitle id="alert-dialog-slide-title">{I18n.get('addTagDialogTitle')}</DialogTitle>
      <DialogContent>
        <DialogContentText id="alert-dialog-slide-description">
          {I18n.get('attachTagDialogContent')}
        </DialogContentText>
        {tags && workflows && (
          <div>
            {Object.keys(tags).filter(tagId => !existedTagIds.includes(tagId)).map(tag => (
              <Chip
                label={`${tags[tag][0]}`}
                key={`${tags[tag][0]}${tags[tag][1]}`}
                name={`${tag}`}
                clickable
                onClick={handleTagClick}
                style={{ backgroundColor: `${tags[tag][1]}`, margin: '0 10px 5px 0', color: 'white' }}
              />
            ))}
          </div>
        )}
      </DialogContent>
      <DialogActions>
        <Button onClick={handleClose} color="primary">
          {I18n.get('close')}
        </Button>
      </DialogActions>
    </Dialog>
  );
};
AttachTagDialog.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  workflowId: PropTypes.string,
  workflows: PropTypes.objectOf(PropTypes.any),
  handleClose: PropTypes.func.isRequired,
  updateTagFromWorkflow: PropTypes.func.isRequired,
  tags: PropTypes.objectOf(PropTypes.array),
};
AttachTagDialog.defaultProps = { tags: null, workflowId: null, workflows: null };
const mapStateToProps = state => ({
  tags: state.tags,
  workflows: state.workflows,
});
const mapDispatchToProps = dispatch => ({
  updateTagFromWorkflow:
    (workflowId, tagIds) => dispatch(updateTagFromWorkflowAction(workflowId, tagIds)),
});
export default connect(mapStateToProps, mapDispatchToProps)(AttachTagDialog);
