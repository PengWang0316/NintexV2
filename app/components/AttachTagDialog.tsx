import React, { memo, useMemo, useCallback } from 'react';
import { connect } from 'react-redux';
import {
  Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle, Slide, Chip,
} from '@material-ui/core';
import { ChipProps } from '@material-ui/core/Chip';
import I18n from '@kevinwang0316/i18n';

import { updateTagFromWorkflow as updateTagFromWorkflowAction } from '../store/Workflows/actions';
import { TagsType } from '../store/Tags/types';
import { Workflows } from '../store/Workflows/types';
import { AppState } from '../store/ConfigureStore';

// Use a customized interface to add name prop to the Chip
interface CustomizedChipType extends ChipProps {
  name: string;
}

// Defind a new wrapper for the Chip component
const CustomizedChip: React.ElementType = (props: CustomizedChipType) => <Chip {...props} />;


interface Props {
  isOpen: boolean;
  tags: TagsType;
  workflowId: string;
  workflows: Workflows;
  handleClose: (event: React.MouseEvent) => void;
  updateTagFromWorkflow: (workflowId: string, tagIds: string) => void;
}

const Transition = React.forwardRef((props, ref) => <Slide direction="down" ref={ref} {...props} />);

export const AttachTagDialog = ({
  isOpen, handleClose, tags = null, workflowId = null, workflows = null, updateTagFromWorkflow,
}: Props) => {
  const existedTagIds = useMemo(() => (Object.keys(workflows.data).length !== 0 && workflowId && workflows.data[workflowId].tags ? workflows.data[workflowId].tags.split(',') : []), [workflows.data]);
  const handleTagClick = useCallback((event: React.MouseEvent) => {
    const target: HTMLElement = event.target as HTMLElement;
    updateTagFromWorkflow(
      workflowId,
      `${workflows.data[workflowId].tags ? `${workflows.data[workflowId].tags},` : ''}${target.tagName === 'span' ? (target.parentNode.parentNode as HTMLElement).getAttribute('name') : (target.parentNode as HTMLElement).getAttribute('name')}`,
    );
  }, [workflows.data]);

  return (
    <Dialog
      open={isOpen}
      TransitionComponent={Transition as React.ComponentType}
      keepMounted
      maxWidth="md"
      onClose={handleClose}
      aria-labelledby="alert-dialog-slide-title"
      aria-describedby="alert-dialog-slide-description"
    >
      <DialogTitle id="alert-dialog-slide-title">{I18n.get('addTagDialogTitle')}</DialogTitle>
      <DialogContent>
        <DialogContentText id="alert-dialog-slide-description">
          {I18n.get('attachTagDialogContent')}
        </DialogContentText>
        {tags && workflows && (
          <div>
            {Object.keys(tags).filter(tagId => !existedTagIds.includes(tagId)).map(tag => (
              <CustomizedChip
                label={`${tags[tag][0]}`}
                key={`${tags[tag][0]}${tags[tag][1]}`}
                name={`${tag}`}
                clickable
                onClick={handleTagClick}
                style={{ backgroundColor: `${tags[tag][1]}`, margin: '0 10px 5px 0', color: 'white' }}
              />
            ))}
          </div>
        )}
      </DialogContent>
      <DialogActions>
        <Button onClick={handleClose} color="primary">
          {I18n.get('close')}
        </Button>
      </DialogActions>
    </Dialog>
  );
};
const mapStateToProps = (state: AppState) => ({
  tags: state.tags,
  workflows: state.workflows,
});
const mapDispatchToProps = dispatch => ({
  updateTagFromWorkflow: (
    workflowId: string, tagIds: string,
  ) => dispatch(updateTagFromWorkflowAction(workflowId, tagIds)),
});
export default connect(mapStateToProps, mapDispatchToProps)(memo(AttachTagDialog));
