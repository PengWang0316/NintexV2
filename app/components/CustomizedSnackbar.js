import React from 'react';
import PropTypes from 'prop-types';
import { Close as CloseIcon, CheckCircle as CheckCircleIcon } from '@material-ui/icons';
import {
  IconButton, Fade, Snackbar, SnackbarContent,
} from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  icon: {
    fontSize: 20,
  },
  iconVariant: {
    opacity: 0.9,
    marginRight: theme.spacing(1),
  },
  message: {
    display: 'flex',
    alignItems: 'center',
  },
});

export const CustomizedSnackbar = ({
  open, handleClose, backgroundColor, content, duration, classes,
}) => (
  <Snackbar
    open={open}
    onClose={handleClose}
    TransitionComponent={Fade}
    autoHideDuration={duration}
  >
    <SnackbarContent
      style={{ backgroundColor }}
      message={(
        <span className={classes.message}>
          <CheckCircleIcon className={classes.iconVariant} />
          {content}
        </span>
      )}
      action={[
        <IconButton
          key="close"
          color="inherit"
          onClick={handleClose}
        >
          <CloseIcon className={classes.icon} />
        </IconButton>,
      ]}
    />
  </Snackbar>
);
CustomizedSnackbar.propTypes = {
  open: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  backgroundColor: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  duration: PropTypes.number,
};
CustomizedSnackbar.defaultProps = { duration: 3000 };
export default withStyles(styles)(CustomizedSnackbar);
