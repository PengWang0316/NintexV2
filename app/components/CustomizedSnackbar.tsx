import React, { ReactElement } from 'react';
import { Close as CloseIcon, CheckCircle as CheckCircleIcon } from '@material-ui/icons';
import {
  IconButton, Fade, Snackbar, SnackbarContent,
} from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

interface Props {
  open: boolean;
  handleClose: (event: React.MouseEvent) => void;
  backgroundColor: string;
  content: string;
  duration?: number;
}

const useStyles = makeStyles(theme => ({
  icon: {
    fontSize: 20,
  },
  iconVariant: {
    opacity: 0.9,
    marginRight: theme.spacing(1),
  },
  message: {
    display: 'flex',
    alignItems: 'center',
  },
}));

export const CustomizedSnackbar = ({
  open, handleClose, backgroundColor, content, duration = 3000,
}: Props): ReactElement => {
  const classes = useStyles({});
  return (
    <Snackbar
      open={open}
      onClose={handleClose}
      TransitionComponent={Fade}
      autoHideDuration={duration}
    >
      <SnackbarContent
        style={{ backgroundColor }}
        message={(
          <span className={classes.message}>
            <CheckCircleIcon className={classes.iconVariant} />
            {content}
          </span>
        )}
        action={[
          <IconButton
            key="close"
            color="inherit"
            onClick={handleClose}
          >
            <CloseIcon className={classes.icon} />
          </IconButton>,
        ]}
      />
    </Snackbar>
  );
};

export default CustomizedSnackbar;
