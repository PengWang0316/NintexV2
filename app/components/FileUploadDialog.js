import React, { useState, Fragment } from 'react';
import PropTypes from 'prop-types';
import {
  Button, Input, Dialog, DialogActions, DialogContent, Typography,
  DialogTitle, DialogContentText,
} from '@material-ui/core';
import green from '@material-ui/core/colors/green';
import { withStyles } from '@material-ui/core/styles';
import I18n from '@kevinwang0316/i18n';

import CustomizedSnackbar from './CustomizedSnackbar';
import ProgressDialog from './ProgressDialog';
import { uploadWorkflows } from '../actions/WorkflowActions';
import { uploadInstances } from '../actions/InstanceActions';
import { uploadActions } from '../actions/ActionActions';

const styles = {
  uploadTitle: { marginTop: 15 },
  inputDiv: { display: 'flex' },
  uploadBtn: { marginLeft: 15 },
  successBg: { backgroundColor: green[600] },
};

export const FileUploadDialog = ({ open, handleClose, classes }) => {
  const [isOpenProgressDialog, setIsOpenProgressDialog] = useState(false);
  const [isOpenSnackbar, setIsOpenSnackbar] = useState(false);
  const [workflows, setWorkflows] = useState();
  const [instances, setInstances] = useState();
  const [actions, setActions] = useState();

  const handleWorkflowsChange = ({ target }) => setWorkflows(target);
  const handleInstancesChange = ({ target }) => setInstances(target);
  const handleActionsChange = ({ target }) => setActions(target);

  const handleSnakbarClose = () => setIsOpenSnackbar(false);

  const handleWorkflowsParse = (text) => {
    uploadWorkflows(text);
    setIsOpenProgressDialog(false);
    setIsOpenSnackbar(true);
  };

  /* handle click workflows upload button */
  const handleWorkflowsUpload = () => {
    if (workflows) {
      setIsOpenProgressDialog(true);
      const reader = new FileReader();
      reader.onloadend = () => handleWorkflowsParse(reader.result);
      reader.readAsText(workflows.files[0]);
      workflows.value = null;
      setWorkflows(null);
    }
  };

  const handleInstancesParse = (text) => {
    uploadInstances(text);
    setIsOpenProgressDialog(false);
    setIsOpenSnackbar(true);
  };

  /* handle click workflows upload button */
  const handleInstancesUpload = () => {
    if (instances) {
      setIsOpenProgressDialog(true);
      const reader = new FileReader();
      reader.onloadend = () => handleInstancesParse(reader.result);
      reader.readAsText(instances.files[0]);
      instances.value = null;
      setInstances(null);
    }
  };

  const handleActionsParse = (text) => {
    uploadActions(text);
    setIsOpenProgressDialog(false);
    setIsOpenSnackbar(true);
  };

  /* handle click workflows upload button */
  const handleActionsUpload = () => {
    if (actions) {
      setIsOpenProgressDialog(true);
      const reader = new FileReader();
      reader.onloadend = () => handleActionsParse(reader.result);
      reader.readAsText(actions.files[0]);
      actions.value = null;
      setActions(null);
    }
  };

  return (
    <Fragment>
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">{I18n.get('addAccountDialogTitle')}</DialogTitle>
        <DialogContent>
          <DialogContentText>{I18n.get('uploadFileDialogContent')}</DialogContentText>
          <Typography color="inherit" variant="body1" className={classes.uploadTitle}>{I18n.get('uploadWorkflowTitle')}</Typography>
          <div className={classes.inputDiv}>
            <Input
              id="workflows"
              type="file"
              fullWidth
              onChange={handleWorkflowsChange}
              accept=".csv"
            />
            <Button size="small" className={classes.uploadBtn} color="primary" onClick={handleWorkflowsUpload}>Upload</Button>
          </div>
          <Typography color="inherit" variant="body1" className={classes.uploadTitle}>{I18n.get('uploadInstanceTitle')}</Typography>
          <div className={classes.inputDiv}>
            <Input
              id="instances"
              type="file"
              fullWidth
              onChange={handleInstancesChange}
              accept=".csv"
            />
            <Button size="small" className={classes.uploadBtn} color="primary" onClick={handleInstancesUpload}>Upload</Button>
          </div>
          <Typography color="inherit" variant="body1" className={classes.uploadTitle}>{I18n.get('uploadActionTitle')}</Typography>
          <div className={classes.inputDiv}>
            <Input
              id="actions"
              type="file"
              fullWidth
              onChange={handleActionsChange}
              accept=".csv"
            />
            <Button size="small" className={classes.uploadBtn} color="primary" onClick={handleActionsUpload}>Upload</Button>
          </div>
        </DialogContent>
        <DialogActions>
          <Button onClick={handleClose} color="primary">
            {I18n.get('close')}
          </Button>
        </DialogActions>
      </Dialog>
      <ProgressDialog open={isOpenProgressDialog} />
      <CustomizedSnackbar
        open={isOpenSnackbar}
        handleClose={handleSnakbarClose}
        backgroundColor={green[600]}
        content={I18n.get('snackbarMessage')}
      />
    </Fragment>
  );
};
FileUploadDialog.propTypes = {
  open: PropTypes.bool,
  handleClose: PropTypes.func.isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  // addAccount: PropTypes.func.isRequired,
};
FileUploadDialog.defaultProps = { open: false };

/* istanbul ignore next */
// const mapDispatchToProps = { addAccount: addAccountAction };
// export default connect(null, mapDispatchToProps)(AddAccountDialog);
export default withStyles(styles)(FileUploadDialog);
