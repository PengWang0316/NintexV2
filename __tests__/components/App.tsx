import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
// import { render } from '@testing-library/react';

import { App } from '../../app/components/App';

jest.mock('react-router-dom', () => ({
  BrowserRouter: 'Router',
  Route: 'Route',
  Switch: 'Switch',
}));
jest.mock('aws-amplify', () => ({ configure: jest.fn() }));
jest.mock('../../app/components/Navbar', () => 'Navbar');

window.clearInterval = jest.fn();
window.setInterval = jest.fn();

describe('Test App component', () => {
  let useEffect;

  const mockUseEffect = () => {
    useEffect.mockImplementationOnce(fn => fn());
  };

  beforeEach(() => {
    useEffect = jest.spyOn(React, 'useEffect');
    mockUseEffect();
    mockUseEffect();
  });

  const mockedMonitorList = {
    isFetched: false,
    data: {
      workflowId: {
        tenant: 'tenantA', instances: { instanceId: { startTime: 'startTime', endTime: 'endTime', status: 'status' } }, key: 'key', hasFailure: false,
      },
    },
  };
  it('should not call clearInterval', () => {
    shallow(<App monitorList={{ isFetched: false, data: {} }} checkInstanceStatus={jest.fn()} />);
    expect(window.clearInterval).not.toHaveBeenCalled();
  });

  it('should call setInterval and clearInterval', () => {
    shallow(<App monitorList={mockedMonitorList} checkInstanceStatus={jest.fn()} />);
    expect(window.clearInterval).not.toHaveBeenCalled();
    expect(window.setInterval).toHaveBeenCalledTimes(1);
  });
});
