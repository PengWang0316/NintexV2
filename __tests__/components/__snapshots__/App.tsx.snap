// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component should have the same snapshot 1`] = `
<Router>
  <div>
    <Navbar />
    <MenuDrawer />
    <main>
      <Switch>
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          exact={true}
          path="/"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          exact={true}
          path="/workflowTable"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          exact={true}
          path="/monitorCenter"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/signin"
        />
        <Route
          render={[Function]}
        />
      </Switch>
    </main>
  </div>
</Router>
`;
